### PR TITLE
Add logging when an expiring registration is pruned

### PR DESCRIPTION
**Pull request by boyanguber**
**Pull Request check list**

- [ x ] Commit conforms to CONTRIBUTING.md?
- [ x ] Proper tests/regressions included?
- [ x ] Documentation updated? - Not sure if this is necessary for this change

**Affected functionality**
Added logging when registrations are pruned due to expiry - no functional changes.

**Description of change**
When a registration is pruned, it will log a message on the info level containing the SPIFFE ID and Parent ID

**Which issue this PR fixes**
None, but these logs will help our use case at uber

### CODE REVIEW

File: pkg/server/datastore/sqlstore/sqlstore.go [hunk 1]
```diff
@@ -3209,7 +3209,7 @@ func deleteRegistrationEntrySupport(tx *gorm.DB, entry RegisteredEntry) error {
 	return nil
 }
 
-func pruneRegistrationEntries(tx *gorm.DB, expiresBefore time.Time) error {
+func pruneRegistrationEntries(tx *gorm.DB, expiresBefore time.Time, ds *Plugin) error {
```
Comment (azdagron) [inline pkg/server/datastore/sqlstore/sqlstore.go:3212]: Can we pass the logger here instead (e.g. `ds.log`)?

File: pkg/server/datastore/sqlstore/sqlstore_test.go [hunk 2]
```diff
@@ -1359,30 +1363,60 @@ func (s *PluginSuite) TestPruneRegistrationEntries() {
...
+			},
+		},
+	} {
+		tt := tt
+		s.T().Run(tt.name, func(t *testing.T) {
+			err = s.ds.PruneRegistrationEntries(ctx, tt.time)
+			s.Require().NoError(err)
+			fetchedRegistrationEntry, err = s.ds.FetchRegistrationEntry(ctx, createdRegistrationEntry.EntryId)
+			s.Require().NoError(err)
+			s.Equal(tt.expectedRegistrationEntry, fetchedRegistrationEntry)
```
Comment (azdagron) [inline pkg/server/datastore/sqlstore/sqlstore_test.go:1415]: We should not use the Suite functions to do the asserts within the subtest or the testing package will complain. Instead the `*testing.T` passed into the subtest should be used with the functions in the `require` or `assert` packages (e.g. `require.NoError(t, err)`).

File: pkg/server/datastore/sqlstore/sqlstore.go [hunk 3]
```diff
@@ -3219,6 +3219,10 @@ func pruneRegistrationEntries(tx *gorm.DB, expiresBefore time.Time) error {
 		if err := deleteRegistrationEntrySupport(tx, entry); err != nil {
 			return err
 		}
+		ds.log.WithFields(logrus.Fields{
+			telemetry.SPIFFEID: entry.SpiffeID,
```
Comment (azdagron) [inline pkg/server/datastore/sqlstore/sqlstore.go:3223]: Do we also want to log the entry id? That will be needed to identify the exact entry that was pruned.

**Review by azdagron — COMMENTED** [review]
Thanks for opening this @boyanguber. Just a few small comments and then we can take the change.

**Comment by azdagron** [comment]
Also, commits to this repository require DCO signoff.

File: pkg/server/datastore/sqlstore/sqlstore.go (hunk 1 above)
Comment (boyanguber) [reply pkg/server/datastore/sqlstore/sqlstore.go:3212]: Sure, thanks!

File: pkg/server/datastore/sqlstore/sqlstore_test.go (hunk 2 above)
Comment (boyanguber) [reply pkg/server/datastore/sqlstore/sqlstore_test.go:1415]: Got it, thanks!

File: pkg/server/datastore/sqlstore/sqlstore.go (hunk 3 above)
Comment (boyanguber) [reply pkg/server/datastore/sqlstore/sqlstore.go:3223]: Sure, I can add that, thanks!

**Comment by azdagron** [comment]
Looks like this is going to fail integration tests since the log entry will differ when testing the different databases (which reuse the unit test suite against each database dialect and version we support). My suggestion is to refactor the test so that the version log entry isn't asserted.

**Comment by boyanguber** [comment]
> 



> Looks like this is going to fail integration tests since the log entry will differ when testing the different databases (which reuse the unit test suite against each database dialect and version we support). My suggestion is to refactor the test so that the version log entry isn't asserted.

Thanks! Let me try that, was wondering why it was failing

**Review by azdagron — APPROVED** [review]
\o/ thanks

**Comment by boyanguber** [comment]
> \o/ thanks

Thanks for reviewing! This was my first open source contribution ever :)

**Comment by azdagron** [comment]
Thanks for the contribution!

**Pull request merged**